src/Components/Main/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,161 @@
+import {
+  memo,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import cn from 'classnames';
+import { useTodoContext } from '../../../Context/Context';
+import { Todo } from '../../../types/Todo';
+import { TemporaryTodo } from '../TemporaryTodo/TemporaryTodo';
+
+export const TodoItem = memo(() => {
+  const [todoForEdit, setTodoForEdit] = useState<Todo | null>(null);
+  const [editTitle, setEditTitle] = useState('');
```

Comment:
https://github.com/mate-academy/react_todo-app-with-api/pull/1004#discussion_r1436304151 - not fixed 😢